### PR TITLE
Fips 2021 10 20 1 mu jitter

### PR DESCRIPTION
**Pull request by Taffer**
### Issues:
Resolves CryptoAlg-1442

### Description of changes: 
Use `clock_gettime_nsec_np(CLOCK_MONOTONIC_RAW);` for Apple Silicon macOS, the previous technique didn't have a high-enough resolution on ARM.

### Call-outs:
The fix in `main` (which removes the `__MACH__` branch) doesn't work in this old FIPS branch. I have no idea why, hopefully just because the version of JitterEntropy is old here.

### Testing:
`ninja run_test` and all is happy in FIPS and non-FIPS mode on Apple Silicon.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and  the ISC license.

### CODE REVIEW

**Comment by torben-hansen** [comment]
Can you upgrade jitter entropy to a newer version?

**Comment by dkostic** [comment]
> The fix in main (which removes the __MACH__ branch) doesn't work in this old FIPS branch
Why it doesn't work? The fix in main should not have any impact on the CI because in the CI we are testing only x86 macOS which would be covered by the existing `#ifdef __x86_64__`, which means the new code path should not even be reachable (in the CI tests).

**Comment by Taffer** [comment]
> > The fix in main (which removes the **MACH** branch) doesn't work in this old FIPS branch
> 
> Why it doesn't work? The fix in main should not have any impact on the CI because in the CI we are testing only x86 macOS which would be covered by the existing `#ifdef __x86_64__`, which means the new code path should not even be reachable (in the CI tests).

Running locally, the unit tests fail with the same jitter errors as before; works fine with the `__MACH__` bit added back in. If it didn't take 650+ days to get through NIST approval, we could drop this in favour of the new FIPS branch...

**Comment by dkostic** [comment]
> Running locally, the unit tests fail with the same jitter errors as before;

Is this on an M1 mac? I just tried it on my M1 and don't see a failure.

**Comment by Taffer** [comment]
> > Running locally, the unit tests fail with the same jitter errors as before;
> 
> Is this on an M1 mac? I just tried it on my M1 and don't see a failure.

Yeah; let me try again.

**Comment by Taffer** [comment]
> > Running locally, the unit tests fail with the same jitter errors as before;
> 
> Is this on an M1 mac? I just tried it on my M1 and don't see a failure.

Maybe my system is "too fast" or something?

```
[419/423] cd /Users/herborth/git/aws-l...-lc-main/build-fips/ssl/test/bssl_shim
FAILED (TLS-TLS1-RSA_WITH_3DES_EDE_CBC_SHA-LargeRecord)
unexpected failure: local error 'child exited early: signal: abort trap', child error 'signal: abort trap', stdout:

stderr:
CPU Jitter entropy RNG initialization failed.
```

Running macOS 12.6.1 with current (updated yesterday, IIRC) command-line tools. Building with `cmake -G Ninja -DFIPS=TRUE -DBUILD_SHARED_LIBS=TRUE -DCMAKE_BUILD_TYPE=RelWithDebInfo ..` from a sub-directory.

**Comment by dkostic** [comment]
> 

amazing, exact same commands on my mac -- doesn't fail.

**Comment by Taffer** [comment]
> > 
> 
> amazing, exact same commands on my mac -- doesn't fail.

But like this are great. :-P
